### ISSUE TITLE
[Feature Request] Images/Screenshots for each Application in the Description

### ISSUE THREAD

**Issue by Satanarious** [marticliment/WingetUI#180]
It would be nice to have images/screenshots of each package/application on the description page.

I would also like to offer my help with collecting the images or links to the images. It would be a time-consuming task since the package list will be ever-growing. I think this would really help the UI and encourage the exploration of packages.

Here's a concept:

![Screenshot 2022-10-02 032221](https://user-images.githubusercontent.com/42642682/193430058-9c968aa7-6255-4e08-8158-615f8f348ccf.png)

**Comment by martinet101** [comment]
Yes, it'd really improve, but the thing is, from where do we get them

**Comment by Satanarious** [comment]
Most Packages have GitHub links or Websites associated with them which have screenshots on them.For the Packages which don't, images/screenshots would be a google search away. I can create a seperate json file with package ids as key and lists of links to images as values.

**Comment by martinet101** [comment]
If you'd do that (really just the most common), it'd be nice, and screenshots will possibly make its way into WingetUI

**Comment by martinet101** [comment]
I guess the json schema could be somethign like
```json
{
  "winget": {
    "packageId": ["UrlToScreenshot1", "urlToScreenshot2"],
    "packageId2": ["UrlToScreenshot3", "urlToScreenshot4"],
  },
  "scoop": {
    "packageI3": ["UrlToScreenshot5", "urlToScreenshot6"],
    "packageId4": ["UrlToScreenshot7", "urlToScreenshot8"],
  }

}
```

and could be perfectly stored in this same repo.

What do you think?

**Comment by Satanarious** [comment]
I was thinking of doing it exactly the same way.

**Comment by martinet101** [comment]
You can check out this file:
https://github.com/martinet101/WingetUI/blob/main/WebBasedData/screenshot-database.json

I'll add the feature when I can.

**Comment by gerardog** [comment]
Hi! I just saw this issue, and just let me express my two cents...

Manual collection of images is literally an infinite task. At some point humans will get bored and abandon the task.
You may automate the collection, but if you do so, why not instead think of on-the-fly mechanisms to fetch the images. 

- For MsStore apps, you may scrap images from the store itself, using it's store id: 
https://apps.microsoft.com/store/detail/9NTXR16HNW1T

- If the project home page is rooted in GitHub, you can get the [GitHub repository social media preview](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/customizing-your-repositorys-social-media-preview), by fetching the repo and filtering `by <meta property="og:image" content="{image url}"`. This may work even in non-github pages.

Other cases, would need to think about, maybe fetching all images from project homepage and filtering those bigger than 200x200...

**Comment by Satanarious** [comment]
There are a lot of packages that I manually did today. I couldn't find screenshots/images for several packages even after using the google image search.  Now, regarding automation:

- Mostly, all apps from scoop and hundreds of apps from Winget don't exist on the MsStore.
- We can't be sure if the repo owner would have uploaded a social media preview image.
- Besides all that, several apps don't even have homepages linked in the description and are marked as unknown.
- Also, I checked a few project homepages, and some have 1 or 2 while others had no screenshots at all. But I'll have to agree that many had screenshots for the app.

Just because this process has to be from the developer's end, the image attachment part. Since we are dealing with cli's here, that is not the case. So, it has to be either done manually or abandoned entirely. 

I was thinking of doing them by setting a daily goal of a number of apps to be done. Once this is done, since Winget and scoop both will have packages being added to them constantly, then someone can create a script to check what apps have missing screenshots in the JSON file I proposed. This way, missing screenshots can be processed with each release.

**Comment by ppvnf** [comment]
Both manual and automatic ways of collecting images have advantages and disadvantages. As [gerardog](https://github.com/gerardog) explained, there's no guarantee people will collect images for new packages and as [Satanarious](https://github.com/Satanarious) said, automation is likely to fail for several packages. So, why not have a manual collection of images with an automatic image fetcher as a fallback?

**Comment by Satanarious** [comment]
How do you propose this image fetcher will differentiate between an actually valuable image (screenshot of the package) and random images (non-useful images) that show up when doing a general google image search

**Comment by Satanarious** [comment]
Update: I'm also doing the icons now along with the screenshots
here's the format:
```json
{
"winget": {
              "icon":"icon image linkl(mostly transparent background png icons)",
              "images": [
                       "image link 1",
                       "image link 2",
                       ]
          }
"scoop": {
              "icon":"icon image linkl(mostly transparent background png icons)",
              "images": [
                       "image link 1",
                       "image link 2",
                       ]
          }
}
```

**Comment by martinet101** [comment]
> How do you propose this image fetcher will differentiate between an actually valuable image (screenshot of the package) and random images (non-useful images) that show up when doing a general google image search

That won't be really possible, so when possible the pics will only bé talen from oficial websites

**Comment by martinet101** [comment]
And i'd add also some placeholder text like "No media found\nDo you want to add screenshots to this package?" witn an url so people notice they can easily contribute adding media to the packages they want

**Comment by martinet101** [comment]
![image](https://user-images.githubusercontent.com/53119851/196262749-4a8e7d49-2cfe-42c0-8fce-f79281d22b5f.png)
👀

**Comment by Satanarious** [comment]
Looks promising, Great work!

**Comment by martinet101** [comment]
![image](https://user-images.githubusercontent.com/53119851/196270161-c25cd3f1-bd71-4ff6-9809-dcd504ae8a6f.png)

Light mode aspect

**Comment by Satanarious** [comment]
> ![image](https://user-images.githubusercontent.com/53119851/196454869-9c6df5bc-7363-4ffe-b80b-cbb2b0c2326d.png)
> And finally, a detailed view of the clicked icons

Maybe add arrows on each side to switch to preceding and succeeding images respectively.

**Comment by martinet101** [comment]
> Edit: Also, keyboard shortcut to switch images through arrow keys would be helpful as well, making the design more intuitive.

This will be discussed in #275

**Comment by martinet101** [comment]
The implementation has been done and the branch will be merged into main

**Comment by Satanarious** [comment]
I figured out that the official websites for the package managers have icons from their respective packages, maybe fetch: 
- Chocolatey icons from https://community.chocolatey.org/
- Scoop icons from https://scoop.sh/

I saw your [issue](https://scoop.sh/) @marticliment on the winstall.app repo. No words from the dev still.

**Comment by marticliment** [comment]
Do you know if they have any kind of api?

**Comment by Satanarious** [comment]
I don't know of any.

Though chocolatey has dedicated web pages for its packages, so scraping off using the package id should not be hard in that case.

In the case of scoop, they have several packages listed on one web page, instead of dedicated web pages for each. You'll have to find a way to scrape those.

Well those are my thoughts, if you somehow consider scraping.

**Comment by marticliment** [comment]
Chocolatey seems yo have static urls for icons, so it seems that with chocolatey it works. Gotta find how scoop works now

![image](https://user-images.githubusercontent.com/53119851/215789920-b889b4df-fe10-43b8-ae01-2478274b7134.png)

**Comment by marticliment** [comment]
Hi @Satanarious, scoop does not provide a proper icon parser, so it won't be implemented in WingetUI. Chocolatey, though, will have icons with the next release
